### PR TITLE
Drag & drop ordering + general views updates

### PR DESCRIPTION
**Pull request by elia**
_A reprise of #15_
_Closes #15_

### CODE REVIEW

**Comment by aldesantis** [comment]
@elia considering that we're moving towards a complete separation of BE and FE, do you think this still makes sense to do? Custom frontends may have completely different logic for displaying pages and not rely on sorting at all.

@kennyadsl what do you think?

File: app/views/spree/admin/pages/edit.html.erb [hunk 1]
```diff
@@ -1,13 +1,35 @@
...
+<% admin_breadcrumb(@page.title) %>
+
+<% content_for :page_title do %>
+  <%= Spree.t(:editing_page) %>
+<% end %>
 
-<% content_for :page_actions do %><% end %>
+<% content_for :page_actions do %>
+  <li>
+    <%= button_link_to I18n.t("spree.back_to_pages"), spree.admin_pages_path, icon: 'arrow-left' %>
```
Comment (spaghetticode) [inline app/views/spree/admin/pages/edit.html.erb]: I would remove the `I18n` namespace from views so we can leverage the more feature loaded version from`ActionView::Helpers::TranslationHelper`

**Comment by elia** [comment]
@aldesantis that's true, although the feature was already there with an half-baked poor man's version of acts_as_list. Adding drag-n-drop on top of that is the best way to let users that want to reorder their pages do so.

File: app/views/spree/admin/pages/edit.html.erb (hunk 1 above)
Comment (elia) [reply app/views/spree/admin/pages/edit.html.erb]: @spaghetticode good call, I'll address this separately in a repo-wide PR 👍

**Comment by kennyadsl** [comment]
I'm ok with this change, which I consider a bugfix. The database column was already there and we are not changing or limiting how users may sort the pages in their frontend, right?

**Comment by elia** [comment]
@kennyadsl that's correct, the only relevant change is the "position" column default, from `false` to `0`.

File: app/models/spree/page.rb [hunk 2]
```diff
@@ -39,20 +34,9 @@ def meta_title
 
 private
 
-  def update_positions_and_slug
+  def check_slug
```
Comment (aldesantis) [inline app/models/spree/page.rb]: I wonder if this method should be called `check_slug`, considering that it actually alters the slug? Wouldn't `normalize_slug` be a better name?

**Pull request merged**